src/wallet/walletdb.h

```diff
@@ -46,9 +46,12 @@ class CHDChain
 {
 public:
     uint32_t nExternalChainCounter;
+    uint32_t nInternalChainCounter;
     CKeyID masterKeyID; //!< master key hash160
 
-    static const int CURRENT_VERSION = 1;
+    static const int VERSION_HD_BASE        = 1;
+    static const int VERSION_HD_CHAIN_SPLIT = 2;
+    static const int CURRENT_VERSION        = VERSION_HD_CHAIN_SPLIT;
```

Comment:
It would upgrade to the new `CHDChain` serialisation format (including the `nInternalChainCounter`), but it would not upgrade to use internal keys, right?
Is that problematic? If so, I can add a fix that keeps the serialisation version of `CHDChain` during encryption.